From a4a3da9c262e4e23059b154cc7546c3a47d53489 Mon Sep 17 00:00:00 2001
From: Jakub Jelen <jjelen@redhat.com>
Date: Mon, 26 Oct 2026 13:17:25 +0100
Subject: [PATCH] Verify source signatures with openpgpverify

Replace the GnuPG-based %gpgverify with %openpgpverify, which uses
Sequoia sqv and takes the same options, as part of the Fedora 45 Change
https://fedoraproject.org/wiki/Changes/Sequoia_openpgpverify

Refresh kdudka.pgp from keyservers/WKD: same certificate(s), with self-signatures our copy was missing.
Release bumped with a changelog entry.

Assisted-by: Claude Opus 5.5
---
 cscppc.spec |   8 ++-
 kdudka.pgp  | 137 +++++++++++++++++++++++++++++++++++++---------------
 2 files changed, 104 insertions(+), 41 deletions(-)

diff --git a/cscppc.spec b/cscppc.spec
index 9e0ef1e..859b841 100644
--- a/cscppc.spec
+++ b/cscppc.spec
@@ -3,7 +3,7 @@
 
 Name:       cscppc
 Version:    2.2.8
-Release:    2%{?dist}
+Release:    3%{?dist}
 Summary:    A compiler wrapper that runs Cppcheck in background
 
 License:    GPL-3.0-or-later
@@ -25,6 +25,7 @@ BuildRequires: gnupg2
 # against the old version of glibc if it was built against a newer one.
 # Therefore, we link glibc statically.
 BuildRequires: glibc-static
+BuildRequires: openpgpverify
 
 # The test-suite runs automatically trough valgrind if valgrind is available
 # on the system.  By not installing valgrind into mock's chroot, we disable
@@ -77,7 +78,7 @@ This package contains the csmatch compiler wrapper that runs the Smatch analyzer
 in background fully transparently.
 
 %prep
-%{gpgverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
+%{openpgpverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
 %autosetup
 
 %build
@@ -139,6 +140,9 @@ done
 %{_mandir}/man1/csmatch.1*
 
 %changelog
+* Thu Oct 22 2026 Jakub Jelen <jjelen@redhat.com> - 2.2.8-3
+- Verify upstream source signatures with openpgpverify
+
 * Wed Jul 15 2026 Fedora Release Engineering <releng@fedoraproject.org> - 2.2.8-2
 - Rebuilt for https://fedoraproject.org/wiki/Fedora_45_Mass_Rebuild
 
diff --git a/kdudka.pgp b/kdudka.pgp
index ee89e90..adb591c 100644
--- a/kdudka.pgp
+++ b/kdudka.pgp
@@ -1,6 +1,6 @@
 -----BEGIN PGP PUBLIC KEY BLOCK-----
 
-mQINBFgjU54BEACwGTSIP9AVBahlfv/y4snLRvlU4UWWqn8bxjh/GFTVs+l8gqOD
+xsFNBFgjU54BEACwGTSIP9AVBahlfv/y4snLRvlU4UWWqn8bxjh/GFTVs+l8gqOD
 3dT9AhbnMWfvr94nA6dXVVx8t8akn3ybVLKeii3vOSel8ayAnIXYjtowPh/TlheO
 BSo4EcVo0IFLtiUhC0XHMngITkr6mGphzKOAjS5Kur1j09tawhWMtgeDWw9dZnvc
 mH7f03mwvFv49YYqztaKcGvWlrLjj1O18Un5euGx18L+udG3RfeWMpzinwvcv2n7
@@ -11,42 +11,101 @@ E61RI+7SJKr4yeboWSfYJbV6RQJyu8X77H9L0F6O+LHoLSoHIRmkcniwEMwl5THV
 tUl9Daxgey+qNq1twLLV6vx8f8eyuPCdeP6ZhhUhOH4sAyh0oGZMHxiNhAFeyRdo
 JqTXfgqLX39jwH54eJ3Cbhndwu47glipMO1HQX1XS5Rt7LfEMCTLUGSFW1xljLOI
 8d9fExEyTzJMVIsQJoaAvPEX4cfhcAUFQLijPkt29Wvv3WsAIVFEgoLMNwARAQAB
-tB9LYW1pbCBEdWRrYSA8a2R1ZGthQHJlZGhhdC5jb20+iQJUBBMBCAA+AhsDBQsJ
-CAcCBhUICQoLAgQWAgMBAh4BAheAFiEEmSqW4HUFbnnNghT5hz2zdXKjezYFAl+1
-eU8FCQ8W87EACgkQhz2zdXKjezaYpw//UwiegIs8Xe79CERudpz7AM0BbRE6VaAU
-QP1dMsTzIUU3HqpRrRfuCLIcbbUb7lCzAmu0SShvrt1ZUY87RXZQDJFsbHneHIKb
-wIxIr6bRtwv1+I9A6bIWYDPdjgost4v2O2GdvDegdC6aDFJa6p7uYF3YqR1GvlCN
-RC0DPvoZLIaHO7q+9o9WN6pe1OBmHdkzfJue9FmJxUhXGhaFGNQ/E9ahZRWv7D4e
-3fxH8B2lqgmLGAYsbMjgiOJFxcbIWMzltIj0hJ1x3ajUdY1B6rLf6QcgXnKJIXVR
-Svp0s283PfhnCzoXvKFvBuUaXQfNsW3MnIJFJEWDuy1TzMdK44AmQp8iQTGVIajd
-2Wdmxxd54dl3GjuHPXXJZ92DG5H52cC+4TZuM4yH9gvOxwtdIafOSkvtTHYh4POF
-piqiM67UG2a8JkW7CKPGFqfrdkM+yOfU31ouHL68q3XIpkB4z1f2w6mscdW2d7AQ
-3VLpb+WCeoWRy6HrRYAJZjs78Rea8N9dSzUOI2ac2OUR9Mqp6TMXed6V+6b1ogbI
-4I0Ni8562kPFxnjiTUhrcXNroBvQUktkEXjuk5ZOG/fJaL0lN39Cq9ImznCEGuvn
-mb+sZ//kH7N5w8tTc3mK4NvQw8LkDyS5LItx1H2Gzybxsl5d0OajJpUY4PZeppjH
-rxXke/QpXHq5Ag0EWCNTngEQALkRI0PUaVE9j19uyjINlxb/3nwKHmbTChQzPJFn
-adUwbmXfChmK/vyE8XBaIFIWSJ/94W9Y1/aGPlK4my7GqkiS4q6Lf32YWBNqihvH
-mxKuIYv2+6Z8E34yRFwmbA20RpZCy7AGIg0/LACfM4Bw+DVUhTRMl2O/muKrxd/O
-/WLn30RoYG+D4+mE0xJu+XsHivx2DqvdkKO+Rzo8131ByiWOk6P37McFtYiPjEjh
-ztTBcnNjd+a3xB/XDHd1Lcs7GmBqw0X10KnxC8xSzSqGSRFYF1aJYdxhayxXGJz/
-p1Dd6mt2eT46rYUGhFWlFH7FXGsWapR8ELY42clcFgGmQ7Yps+dZ6Kx8HnEYKsIY
-ONBqjS/dTKSrOMvkCSY0CwiCjKPM5uan5lQ9GMwbEZOQ5dcEVJOiVSfneeYpEjD/
-oyapPrDefdsCD5Gvt2kSbDZSDR5GeO8epZ02hu/zMQxDayqdLTxAaDByDVTvRCnc
-BLDcpvzXVAUdjIkfzDqZlLRgZu/8oNjOpWypUEE0mQfus6fDOLrt1h/0SqcJar70
-mi0QzBlOLrksJerXygDYJus80trCJPbr5DkCy2nQdfaeUissbt4kJTBirhhMtuyZ
-bBOQ42qm5pGef74hye1dCUddlBcb/BmIecsQ5a7EegKBDoU6ZsLcs5xnPgNwJa5U
-5VstABEBAAGJAjwEGAEIACYCGwwWIQSZKpbgdQVuec2CFPmHPbN1cqN7NgUCX7V5
-agUJDxbzzAAKCRCHPbN1cqN7NiVdEACGZX+sMSfpW47ARmsg9EsWh983SafWEi4V
-Gp3bRgOM3X4hwp8iFS/jpD8iNQpiRztSAx6s0l2pirAKFiKaaHrarVrYM4lrSoau
-J1LeWeAy9jHRstk21Iu/myM8gfBdl9tOlrdv5NhD98tCdE/2hTtOLlZbYboNl+ug
-0g/3yM4KPgqXLvVpS3QBoiueTfFoSawb20lZCcDon43BGg+wS/2j7Vu9Q1Dj3fEz
-+QV4S7JvMFP6MYV2ITvj3xajXpRkuNG8s76o/u8m2PYQ77sAl+mN446Lp+bwdQeE
-s7j79i/2kk+djVDtgTGyRyDD/4drXOMtVKRpxDDp1YOl896cRP4PJWNK8oLlF8IY
-ItdhN/UijK6hZoXLyQDK/DQfmTjpGEQTzFCNW8CdwvTSjK7o6lJZtrv4R4rBJ3Sd
-kcr9rQO/uGlYblzX70iXQMKpiCb1xo3MBCUFfiq05sTNVzRNVleo9nVf0WhCgnl7
-M9Tojh31sra9IzDAy9exga8dD/tvnebYjXYmGXfQyrPAnSSTLSjAQmlNzgx8FM96
-WB+XJDJFALy/MV35XKi9c5SLE3hSPEhqrwnTQ5g3jOPrexhUZR6w0qDXVoQH/3p0
-vXqQ3yx3yrREeBOW6qhHeYk3w2z7EAg4nNovAHgd68zXE9ZfCAGfWIerZsOuhdHS
-lwvfpMesuQ==
-=XhUt
+zR9LYW1pbCBEdWRrYSA8a2R1ZGthQHJlZGhhdC5jb20+wsGUBBMBCAA+AhsDBQsJ
+CAcCBhUICQoLAgQWAgMBAh4BAheAFiEEmSqW4HUFbnnNghT5hz2zdXKjezYFAmcy
+ERoFCRaTi3wACgkQhz2zdXKjezaCNQ//QC6YB/bW5qXOin2OZ/LMr2iqJCGMGDYl
+8+5Xwk0I30bkAo/JlR3u+Nag0BHd783ngKkbwOiJ2vYPPD+XYzjkXHMhXI0MGtAT
+jrHkJ+uOs/ODd9qL6V47MYb/PMyskE5ke0GsXwtGUjc8Xo7ZJZFQykO1bxr9v8ml
+N0eRM6fqWjLlF4Qv+67JKFYjLuz5OJe+8LyQJaexATo10AZMBORru8f/dPLEhEba
+v552mdvZt9e9eFyrcMjdmFC4/XgLWLHSNro+Ec3dcUwm4auwz8oGNb9iu5AaCIsf
+Z+UsMlPBNYmVN5sLypAfriLXKFS5F5dkf8BkqkozmmGVSI9Mvn6h5wn5Z/W+I7am
+F+X18ORJvIIjJ2xYEI8dBg8b3pdqajT4+PBXUpnrgH28ACuieYZ4jRGDbh5wC2//
+vwGCMHvf1efKM5F7jksKA0owA9TxtyGkx6lW1vZ7j/SkKC6+ZYkFPCgQv0gcTUSW
+T3z7BU6knvs3JljTeQI8VJgSYsjP1B7es6gAhkefYcQnHe+gZCdaqjK9ZpEoJ0Ia
+R/Wy0DEoaoXoJ9R6uvMycIfhGNAWp3YmF6HQsAHtjCZLyFFgjsg+uIP0TuPvORvB
+aa3yZXbALoSC3ZX4Mvmy08p6z+QFlHRAEYWcVWAZaWCgAwdkLeyFcU5yKHuU1RZg
+1mE4AYfHKSXCwZQEEwEIAD4CGwMFCwkIBwIGFQgJCgsCBBYCAwECHgECF4AWIQSZ
+KpbgdQVuec2CFPmHPbN1cqN7NgUCX7V5TwUJDxbzsQAKCRCHPbN1cqN7NpinD/9T
+CJ6Aizxd7v0IRG52nPsAzQFtETpVoBRA/V0yxPMhRTceqlGtF+4IshxttRvuULMC
+a7RJKG+u3VlRjztFdlAMkWxsed4cgpvAjEivptG3C/X4j0DpshZgM92OCiy3i/Y7
+YZ28N6B0LpoMUlrqnu5gXdipHUa+UI1ELQM++hkshoc7ur72j1Y3ql7U4GYd2TN8
+m570WYnFSFcaFoUY1D8T1qFlFa/sPh7d/EfwHaWqCYsYBixsyOCI4kXFxshYzOW0
+iPSEnXHdqNR1jUHqst/pByBecokhdVFK+nSzbzc9+GcLOhe8oW8G5RpdB82xbcyc
+gkUkRYO7LVPMx0rjgCZCnyJBMZUhqN3ZZ2bHF3nh2XcaO4c9dcln3YMbkfnZwL7h
+Nm4zjIf2C87HC10hp85KS+1MdiHg84WmKqIzrtQbZrwmRbsIo8YWp+t2Qz7I59Tf
+Wi4cvryrdcimQHjPV/bDqaxx1bZ3sBDdUulv5YJ6hZHLoetFgAlmOzvxF5rw311L
+NQ4jZpzY5RH0yqnpMxd53pX7pvWiBsjgjQ2LznraQ8XGeOJNSGtxc2ugG9BSS2QR
+eO6Tlk4b98lovSU3f0Kr0ibOcIQa6+eZv6xn/+Qfs3nDy1NzeYrg29DDwuQPJLks
+i3HUfYbPJvGyXl3Q5qMmlRjg9l6mmMevFeR79ClcesLBfQQTAQgAJwUCWCNTngIb
+AwUJB4TOAAULCQgHAgYVCAkKCwIEFgIDAQIeAQIXgAAKCRCHPbN1cqN7Nl5uD/9m
+yhvNK09HuZRcLLWqprtOORpG9SfcLagthxc6Seg80cHkqR7rc8se+yEHe/+34xrc
+3952XIS5yfc+QaRv0lQKtjuiheQfPVO7B57vOE1SpNEURL+5fk+ZZocuXDr3RY0D
+PfAFfR7tTUm/OUp/6pjgWDlUhHhkuW75D0/ozthjNY75Du23X/dUqXsS848zruFb
+kwrzUuqs8LhUN0AF9HiZKEIoa51Q+Ix12R12KrboC3d0FF+gPmGLpZudSw/Q0ND9
+KZ4APl6rtH8WuvujWb0DZfBb1NMPFLzIA8w5DAQOxC+CvxKbgUpywF77AoZT9Wx2
+/d80vMPkMcPwtKsI6sNGnPbbFX0aLmz+9aUytEQQZf9vav5yeGHrtskxRdd3Cw5N
+JpmVFwx4U8MV1KT/wW/66IPGwqH/4GxGq57NsVTAsIgeGrUSuxPLlK6GgMtm/GNQ
+npbYEQnhyvgBHGKTnUBsiDJ8E9lwrtWFAFwKKifpIx6tM+7CqPDDRjWs0mSSu1XR
+SFLCn/ymLKE4yw7sGDvB1gzjs4EvW7JmLtzWnT9DLd0AKO2v9oSJCigxZjYiZwCp
+U1dOcc6LvB1VVx1PgtIP0plclsn0/Fc411fXSwA3BZ2/FfCCtXYxL5xY3bAM8ekU
+kM/Hm/1t7KRqoKzD7Z1VUhXQYNu56kKjUyRihKC4UcLBXAQQAQIABgUCWEF6WQAK
+CRB3H+3RQJWAq3GPD/4waooSUrVFIu3VvS9W9dbKC+fL4YHMZMGEJn2BeQgv0BN1
+/Y9vG4IwYWw4qI8LvMguNvhn2eGIVBwYA3/w+Ia8saOz5++7Y1gMUgat0c4vS2hk
+TnUDh+pRgoc4L6PSZgLAHTdEG9ek07Rksh0eJV2iVtQ8zIIRP+sWrz1O+IrRMQXB
+GkVTno3Dt3+EB6Dj+det9szU1G3B1TUPgUhyEbl8eikPi2+TmtHRcUEL2N6yOXCd
+mcs/OBBmWTz721NsTPGARDsoagZJ30QJHcGz3fFtAWjpK3R9YPARLbmDpP/avIGK
+I+vjTrYnNO7JG1vsFmhL4YTRkS4WEi+DinoDPXOcLqJ725YAG4gT7rpyUh91GLTR
+yXMy5wd9wxwYdirupqOvj5ZOzJyHwruFhhFBUNLV6fVs5w8FSeIVsFGIkw3YTv0j
+QmZMYm1ps5zCv7It3cF/pKdYrKJiBma3YCanZRq7j2W0sBoijjr/Y+Wc+sbPaSVv
+GUYmNPgIe/QraZZWq4cxWXkP4r+ABVn0BFPwkzbJtQUZ/G6wRy6AzCLxH+0KeRqM
+FZE0bxCf8uApuOYbYrXAMg0HpKJmlCirf8sTIX/PS1CR6iuoIwkMMA0S8PcF+its
+sIK8GlmGav8zx9uL3wkA7fKeIxDgn3gAn2y8TgJhvbFVMM1vnSDy6QBYjC+Op87B
+TQRYI1OeARAAuREjQ9RpUT2PX27KMg2XFv/efAoeZtMKFDM8kWdp1TBuZd8KGYr+
+/ITxcFogUhZIn/3hb1jX9oY+UribLsaqSJLirot/fZhYE2qKG8ebEq4hi/b7pnwT
+fjJEXCZsDbRGlkLLsAYiDT8sAJ8zgHD4NVSFNEyXY7+a4qvF3879YuffRGhgb4Pj
+6YTTEm75eweK/HYOq92Qo75HOjzXfUHKJY6To/fsxwW1iI+MSOHO1MFyc2N35rfE
+H9cMd3UtyzsaYGrDRfXQqfELzFLNKoZJEVgXVolh3GFrLFcYnP+nUN3qa3Z5Pjqt
+hQaEVaUUfsVcaxZqlHwQtjjZyVwWAaZDtimz51norHwecRgqwhg40GqNL91MpKs4
+y+QJJjQLCIKMo8zm5qfmVD0YzBsRk5Dl1wRUk6JVJ+d55ikSMP+jJqk+sN592wIP
+ka+3aRJsNlINHkZ47x6lnTaG7/MxDENrKp0tPEBoMHINVO9EKdwEsNym/NdUBR2M
+iR/MOpmUtGBm7/yg2M6lbKlQQTSZB+6zp8M4uu3WH/RKpwlqvvSaLRDMGU4uuSwl
+6tfKANgm6zzS2sIk9uvkOQLLadB19p5SKyxu3iQlMGKuGEy27JlsE5DjaqbmkZ5/
+viHJ7V0JR12UFxv8GYh5yxDlrsR6AoEOhTpmwtyznGc+A3AlrlTlWy0AEQEAAcLB
+fAQYAQgAJgIbDBYhBJkqluB1BW55zYIU+Yc9s3Vyo3s2BQJnMhFGBQkWk4uoAAoJ
+EIc9s3Vyo3s2jhUP/RIoK0a1HCei2vtTlSrqZrW2dCubvkmCU8wzWkYPLC17z+hT
+59Inf8/w3aVHozFko4ZU+ucvtIZ7m4sR7Y2qoAmRitWW2UQ86u+zw1ob7I1t7UNm
+F+oF4q5IeFbnMgJ2tNzBvGOw6WyKlND0sSq7ZfZITVXP/DUZ5eYIyzcxygpdj1Or
+2v65IshsGmdB3gguM7qRDjxj1PbkjITaRgfBDMLbMqSk8AC/PLmT1qrEVMJcpE9O
+rbXcyEjEpKu4qTyhpTDJBIKED6HC/OmLqYzunaJjEjKxydPmJF/IOAhSpq540psi
+YE/qyX0KwzTHNI+yBXv3kBElzW/dFv/iAiL7TaJiQsUgqI442RR6Ost295/ZaSjc
+2VE2aN1PcxquhnfSVHU7sjpk4qS4wjNGmlIv5ynKTRLfZAHz+rlLMCmuiJGoxgNe
+OUrE5WaglmsGd+f7rDIUDExzAO+mI65cpJ4qD06gPU2jB1JAsUoJ2PWYDp0nEvoY
+wmE2rdg4fNJnNcztkCNpW2rHTeGjimWBMPINSrJoaGxHPSdmYwllXxf9bjAKVOu3
+/bVhNc5czXmoCE8kf3wcj08j5btWlzfh6ArIAYXzu+KavbTP/NH4IzF73L2ixV6O
+CCCr25gDNIKwY4tUhiy6ojMEiSF0FDQ9vTQb/MfLV+KKg8fYsWB4/T+ddqj+wsF8
+BBgBCAAmAhsMFiEEmSqW4HUFbnnNghT5hz2zdXKjezYFAl+1eWoFCQ8W88wACgkQ
+hz2zdXKjezYlXRAAhmV/rDEn6VuOwEZrIPRLFoffN0mn1hIuFRqd20YDjN1+IcKf
+IhUv46Q/IjUKYkc7UgMerNJdqYqwChYimmh62q1a2DOJa0qGridS3lngMvYx0bLZ
+NtSLv5sjPIHwXZfbTpa3b+TYQ/fLQnRP9oU7Ti5WW2G6DZfroNIP98jOCj4Kly71
+aUt0AaIrnk3xaEmsG9tJWQnA6J+NwRoPsEv9o+1bvUNQ493xM/kFeEuybzBT+jGF
+diE7498Wo16UZLjRvLO+qP7vJtj2EO+7AJfpjeOOi6fm8HUHhLO4+/Yv9pJPnY1Q
+7YExskcgw/+Ha1zjLVSkacQw6dWDpfPenET+DyVjSvKC5RfCGCLXYTf1IoyuoWaF
+y8kAyvw0H5k46RhEE8xQjVvAncL00oyu6OpSWba7+EeKwSd0nZHK/a0Dv7hpWG5c
+1+9Il0DCqYgm9caNzAQlBX4qtObEzVc0TVZXqPZ1X9FoQoJ5ezPU6I4d9bK2vSMw
+wMvXsYGvHQ/7b53m2I12Jhl30MqzwJ0kky0owEJpTc4MfBTPelgflyQyRQC8vzFd
++VyovXOUixN4UjxIaq8J00OYN4zj63sYVGUesNKg11aEB/96dL16kN8sd8q0RHgT
+luqoR3mJN8Ns+xAIOJzaLwB4HevM1xPWXwgBn1iHq2bDroXR0pcL36THrLnCwWUE
+GAEIAA8FAlgjU54CGwwFCQeEzgAACgkQhz2zdXKjezbBVQ//ecz0iCbSA56vTWN5
+RmDKJ20xlNzvob/UOWQQ9CeTM7If0KaQHdXkTgxZUH139/s7oj8w18QPP/Vr+BWi
+8XZtWCwmGkGwdeWkwnA8AGZ95ulMPYPKAH7q/h0bXDoLNxc6gC92jaJLDqM5XP8P
+PffS5CmZlrxmNBTvEuNOkqeQuZCzfnb1sJZ0Ws+X5h3xvIiqokQZXccg9QHR47Nu
+xzdWwQV5FPEFN0dthjVI0EnA+NVZGAQxR/g9OMjBgadVrv3rVgl4SyMz+Ltb8GWl
+mPuPBv0rXVO9amxUXjCrtcY/c3KQQI3RRVwFvN3xhzbQ3t1fFDoGi2RbIBcEerpm
+GGn9G3nosF6+aLeTHJ78ETcOXiZKELWXLcOcKAGs1rV4xRfQeGSLc+EaElMtqsaP
+ZWJCJUs+waITPPAB6IYz7gt+0x/ibTalom5dRn9IseQZ+DQqGfuonFvpx6ADfUcT
+d0vtoUT6SjaEctsL6PbCGaOJRdU4RgFkYYhwEczuOnwsmA21fSYsbEG2nEyvfcUf
+igrAtfUbJElJnd0a7ofdH6v4Y9m46Je+kpVVElcFh2G5QL0CLZsPJl13z+xQss5c
+Tlo7tT3Wug+d16OtZl5+WmheEHCXu799BwJfsF/qL5jpzeT2aafmDeARypl7/B6R
+AasBOwVgmoA+EuKU0Y4gukEqvfk=
+=JK8P
 -----END PGP PUBLIC KEY BLOCK-----
-- 
2.56.0

